posts/change-data-capture-cloudsql-postgres-with-apache-airflow-1.md

```diff
@@ -0,0 +1,1002 @@
+
+In general, the API call looks like this:
+
+```bash
+POST https://www.googleapis.com/sql/v1beta4/projects/<project-id>/instances/<instance-id>/export
+{
+ "exportContext":
+   {
+      "fileType": "CSV",
+      "uri": "gs://bucket_name/path_to_csv_file",
+      "databases": ["database_name"],
+      "offload": false
+      "csvExportOptions":
+       {
+           "selectQuery":"SELECT Query - no CTE allowed - subquery okay"
+       }
+   }
+}
+```
+
+This will query `database_name` (only one allowed at a time) using the enclosed `selectQuery` and format the results to a CSV at the GCS bucket and keypath set in `uri`. 
+
+  > *Note: This CSV will not have headers, and in turn, you will also run a second operation to obtain the table schema, which will be described below*
+
+## Step 1: Set Up Airflow
+
+The [Astronomer CLI](https://github.com/astronomer/astro-cli) is the easiest way to run Apache Airflow on your machine. From the CLI, you can establish a local testing environment regardless of where you'll be deploying to from there.
+
+There are two ways to install any version of the Astronomer CLI:
+
+- cURL
+- [Homebrew](https://brew.sh/)
+
+### **Prerequisites**
+
+The Astronomer CLI installation process requires [Docker](https://www.docker.com/) (v18.09 or higher).
+
+### **Install with Homebrew**
+
+If you have Homebrew installed, from your terminal run:
+
+```bash
+brew install astronomer/tap/astro
+```
+
+To install a specific version of the Astro CLI, you'll have to specify `@major.minor.patch`. To install v0.16.1, for example, run:
+
+```python
+brew install astronomer/tap/astro@0.16.1
+```
+
+### **Install with cURL**
+
+To install the latest version of the Astronomer CLI, run:
+
+```bash
+curl -sSL https://install.astronomer.io | sudo bash
+```
+
+To install a specific version of the Astronomer CLI, specify `-s -- major.minor.patch` as a flag at the end of the cURL command. To install v0.16.1, for example, run:
+
+```bash
+curl -sSL https://install.astronomer.io | sudo bash -s -- v0.16.1
+```
+
+### **Confirm the Install**
+
+To make sure that you have the Astronomer CLI installed on your machine, run:
+
+```bash
+astro version
+```
+
+If the installation was successful, you should see the version of the CLI that you installed in the output:
+
+```bash
+Astro CLI Version: 0.15.0
+Git Commit: c4fdeda96501ac9b1f3526c97a1c5c9b3f890d71
+```
+
+For a breakdown of subcommands and corresponding descriptions, you can always run `$ astro` or `$ astro --help`.
+
+  >*For more information on installing the Astronomer CLI, please check out the page the [Astronomer CLI Quickstart](https://www.astronomer.io/docs/cloud/stable/develop/cli-quickstart).*
+
+## Step 2: GCP Access - CloudSQL to GCS
+
+To provide access between Airflow, CloudSQL, and a GCS bucket, you first need to give your CloudSQL instance the necessary permissions to make the export to a GCS Bucket, followed by providing Airflow the access to make API calls. 
```

Comment:
"To allow for access between Airflow, CloudSQL, and your GCS bucket, you'll need to both give your CloudSQL instance the necessary permissions to export to the GCS bucket and give Airflow access to make API calls to both services."